CMakeLists.txt

```diff
@@ -93,27 +90,41 @@ else ()
 endif ()
 
 
+target_compile_definitions(
+        betafold PRIVATE SOURCES_ROOT="${SOURCES_ROOT}"
+)
 target_include_directories(betafold
         PRIVATE ${Python_INCLUDE_DIRS}
         ${pybind11_INCLUDE_DIRS}
-        src)
-
+        cbetafold/include
+        cbetafold
+)
 set_target_properties(betafold PROPERTIES
         LIBRARY_OUTPUT_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}"
 )
-
 target_link_libraries(betafold PRIVATE pybind11::module Python::Python)
 
+add_executable(train cbetafold/main.cpp)
+target_compile_definitions(
+        train PRIVATE SOURCES_ROOT="${SOURCES_ROOT}"
+)
+target_include_directories(train
+        PRIVATE ${Python_INCLUDE_DIRS}
+        ${pybind11_INCLUDE_DIRS}
+        cbetafold/include
+        cbetafold)
+set_target_properties(train PROPERTIES
+        LIBRARY_OUTPUT_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}"
+)
+target_link_libraries(train PRIVATE pybind11::module Python::Python)
```

Comment:
This defines definitions for the `train` executable, which should implement `Trainer` to generate data and do the training.